### ISSUE TITLE
Error installing curb on OSX

### ISSUE THREAD

**Issue by jiecaranda** [taf2/curb#285]
When I tried to install curb with command 
`gem install curb`

I got the following message:

make "DESTDIR=" clean

make "DESTDIR="
compiling curb.c
In file included from curb.c:8:
In file included from ./curb.h:12:
In file included from /Users/yujie/anaconda/include/curl/curl.h:35:
/Users/yujie/anaconda/include/curl/curlrules.h:143:6: error: '**curl_rule_01**' declared as an array with a negative size
    [CurlchkszEQ(long, CURL_SIZEOF_LONG)];
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/yujie/anaconda/include/curl/curlrules.h:132:27: note: expanded from macro 'CurlchkszEQ'
# define CurlchkszEQ(t, s) sizeof(t) == s ? 1 : -1

```
                      ^~~~~~~~~~~~~~~~~~~~~~~
```

/Users/yujie/anaconda/include/curl/curlrules.h:153:6: error: '**curl_rule_02**' declared as an array with a negative size
    [CurlchkszEQ(curl_off_t, CURL_SIZEOF_CURL_OFF_T)];
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/yujie/anaconda/include/curl/curlrules.h:132:27: note: expanded from macro 'CurlchkszEQ'
# define CurlchkszEQ(t, s) sizeof(t) == s ? 1 : -1

```
                      ^~~~~~~~~~~~~~~~~~~~~~~
```

2 errors generated.
make: **\* [curb.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Library/Ruby/Gems/2.0.0/gems/curb-0.9.1 for inspection.
Results logged to /Library/Ruby/Gems/2.0.0/extensions/universal-darwin-13/2.0.0/curb-0.9.1/gem_make.out

Any help would be appreciated!

**Comment by spachner** [comment]
Hi,

had the same error. Seemed to clutter with 2nd libcurl installation done by MacPorts (port). I uninstalled the port by "sudo port uninstall curl" and gem installed curb successfully.

regards

spachner

**Comment by yuryskaletskiy** [comment]
Got the same issue, @spachner 's advice helps for me. 
```

(tms-scripts) Yury-Mac:~ enox$ sudo port uninstall curl
The following versions of curl are currently installed:
 1) curl @7.50.3_0+ssl
 2) curl @7.54.0_0+ssl (active)
Enter option(s) [1-2/all]: all

(tms-scripts) Yury-Mac:~ enox$ env ARCHFLAGS="-arch x86_64"; pip install pycurl
Successfully installed pycurl-7.43.0
```

**Comment by Nikhil-Wagh** [comment]
I don't have `port` installed. How do I know if the port is cluttered?

**Comment by below** [comment]
I am having the same issue on macOS 10.14.6, and tried curb up to 0.98

I also don't have `port` installed, so I have no clue what is wrong :(

**Comment by anas10** [comment]
Still no solution?

I also get this on catalina 10.15.6

```
Installing curb 0.9.11 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /private/var/folders/dw/n7c79n4946z2x335zzyb86d00000gn/T/bundler20201118-61597-1u6ug7ocurb-0.9.11/gems/curb-0.9.11/ext
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby -I /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0 -r ./siteconf20201118-61597-nwi5kr.rb
extconf.rb
checking for curl-config... yes
checking for curlopt_tcp_keepalive... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/$(RUBY_BASE_NAME)
	--with-curl-dir
	--without-curl-dir
	--with-curl-include
	--without-curl-include=${curl-dir}/include
	--with-curl-lib
	--without-curl-lib=${curl-dir}/lib
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:467:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:585:in `block in try_compile'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:534:in `with_werror'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:585:in `try_compile'
	from extconf.rb:52:in `block in have_constant'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:959:in `block in checking_for'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:361:in `block (2 levels) in postpone'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:331:in `open'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:361:in `block in postpone'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:331:in `open'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:357:in `postpone'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/mkmf.rb:958:in `checking_for'
	from extconf.rb:44:in `have_constant'
	from extconf.rb:62:in `<main>'

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /var/folders/dw/n7c79n4946z2x335zzyb86d00000gn/T/bundler20201118-61597-1u6ug7ocurb-0.9.11/extensions/universal-darwin-19/2.6.0/curb-0.9.11/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /var/folders/dw/n7c79n4946z2x335zzyb86d00000gn/T/bundler20201118-61597-1u6ug7ocurb-0.9.11/gems/curb-0.9.11 for inspection.
Results logged to /var/folders/dw/n7c79n4946z2x335zzyb86d00000gn/T/bundler20201118-61597-1u6ug7ocurb-0.9.11/extensions/universal-darwin-19/2.6.0/curb-0.9.11/gem_make.out

An error occurred while installing curb (0.9.11), and Bundler cannot continue.
```

**Comment by taf2** [comment]
Are you sure you have Xcode on your machine and the compiler is working?

**Comment by anas10** [comment]
Yes, all good now. I fixed this by switching my env to ruby 2.6.5 instead of the default one that comes with macOS.

**Comment by taf2** [comment]
Ah makes sense probably the default ruby env in MacOS doesn't support native gems?
